src/transformers/models/dbrx/modeling_dbrx.py

```diff
@@ -0,0 +1,1439 @@
+            module.w2.data.normal_(mean=0.0, std=std)
+
+    def _setup_cache(
+        self, cache_cls: Any, max_batch_size: int, max_cache_len: int
+    ):  # TODO: how to set var type of class?
+        if self.config._attn_implementation == "flash_attention_2" and cache_cls == StaticCache:
+            raise ValueError(
+                "`static` cache implementation is not compatible with "
+                + "`attn_implementation==flash_attention_2`. Make sure to use "
+                + "`spda` in the mean time and open an issue at https://github.com/huggingface/transformers."
+            )
+
+        for block in self.transformer.blocks:
+            device = block.norm_attn_norm.norm_1.weight.device
+            if hasattr(self.config, "_pre_quantization_dtype"):
+                dtype = self.config._pre_quantization_dtype
+            else:
+                dtype = block.norm_attn_norm.attn.out_proj.weight.dtype
+            block.norm_attn_norm.attn.past_key_value = cache_cls(
+                self.config, max_batch_size, max_cache_len, device=device, dtype=dtype
+            )
+
+    def _reset_cache(self):
+        for block in self.transformer.blocks:
+            block.norm_attn_norm.attn.past_key_value = None
+
+
+class DbrxModel(DbrxPreTrainedModel):
+    """Transformer decoder consisting of *config.num_hidden_layers*
+
+    [`DbrxBlock`] layers.
+
+    Args:
+        config ([`DbrxConfig`]): Model configuration class with all parameters of the model.
+            Initializing with a config file does not load the weights associated with the model, only the
+            configuration. Check out the [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+    """
+
+    def __init__(self, config: DbrxConfig):
+        super().__init__(config)
+        self.padding_idx = config.pad_token_id
+        self.vocab_size = config.vocab_size
+        self.emb_pdrop = config.emb_pdrop
+
+        self.wte = nn.Embedding(config.vocab_size, config.d_model, self.padding_idx)
+        self.blocks = nn.ModuleList([DbrxBlock(config, block_idx) for block_idx in range(config.n_layers)])
+        self.norm_f = nn.LayerNorm(config.d_model, bias=False)
+        self.gradient_checkpointing = False
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self) -> nn.Embedding:
+        return self.wte
+
+    def set_input_embeddings(self, value: nn.Embedding):
+        self.wte = value
+
+    def _autocast_input_embeddings(self, inputs_embeds: torch.Tensor) -> torch.Tensor:
+        if inputs_embeds.device.type == "cuda" and torch.is_autocast_enabled():
+            return inputs_embeds.to(dtype=torch.get_autocast_gpu_dtype())
+        elif inputs_embeds.device.type == "cpu" and torch.is_autocast_cpu_enabled():
+            return inputs_embeds.to(dtype=torch.get_autocast_cpu_dtype())
+        else:
+            return inputs_embeds
```

Comment:
Not sure why this is still here, we usually don't handle this

Reply:
I think this is basically to ensure that the `input_embeds` output by the `nn.Embedding` layer are in low precision, especially in the case where autocast is on but the weights are not in FP32. But since other `transformers` models don't do this, I think we can remove.